### PR TITLE
ADSB: Add Sagetech driver

### PR DESCRIPTION
**Pull request by magicrub**
This adds a driver for the Sagetech ADSB transceiver. The ADSB wiki page is getting a little out of date so I plan to update that as well.

- Created ADSB_Backend
- abstracted MAVink specific work into it's own AP_ADSB_MAVLink driver for uAvionix
- Added Sagetech driver for the XP protocol. There is an MX protocol for newer products that is also available but not yet implemented. This driver is set up prepared for that future protocol and parse for either protocol within this sagetech driver.
- Sagetech ADSB-in is implemented but not tested. I only had ASSB-out hardware to test with.
- general rework of a variety of ADSB related minor improvements
- adds math library of byte parsing to load arrays
- adds math library to convert math bases (example: dec 12345 to 0x12345)

It would be nice if some people could test with uAvionix Ping products just to sanity check that there is no impact on them.

Code change impact on **fmuv1**:
Target         Text    Data  BSS     Total
without PR (master):
bin/arduplane  967604  1172  195636  1164412
with PR:
bin/arduplane  971292  1172  195636  1168100

delta: **3688 program bytes**

### CODE REVIEW

**Comment by magicrub** [comment]
This was failing CI on several unrelated code paths and it would fail in a different place when I retriggered CI. I see there's a few autotest related commits in master so rebased. Fingers crossed!

File: ArduPlane/ArduPlane.cpp [hunk 1]
```diff
@@ -266,7 +266,7 @@ void Plane::one_second_loop()
     // make it possible to change orientation at runtime
     ahrs.update_orientation();
 
-    adsb.set_stall_speed_cm(aparm.airspeed_min);
+    adsb.set_stall_speed_cm(aparm.airspeed_min * 0.8f); // actual stall speed is 80% of this param
```
Comment (WickedShell) [inline ArduPlane/ArduPlane.cpp:269]: Unrelated change, that I'm assuming wasn't meant to come in here? I don't see any way to establish that the stall speed is actually 80% of our minimum airspeed. It's very common for this to be different.

Comment (magicrub) [reply ArduPlane/ArduPlane.cpp:269]: ARSPD_FBW_MIN description:
Minimum airspeed demanded in automatic throttle modes. Should be set to 20% higher than level flight stall speed.

https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/Parameters.cpp#L353

File: libraries/AP_ADSB/AP_ADSB.cpp [hunk 2]
```diff
@@ -158,52 +151,91 @@ const AP_Param::GroupInfo AP_ADSB::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("LOG",  14, AP_ADSB, _log, 1),
 
+    // @Param: TYPE
```
Comment (IamPete1) [inline libraries/AP_ADSB/AP_ADSB.cpp:154]:
```suggestion
    // @Param: PROTO
```

File: ArduPlane/ArduPlane.cpp (hunk 1 above)
Comment (WickedShell) [reply ArduPlane/ArduPlane.cpp:269]: It doesn't actually get used that way, it's more of a minimum then anything else...

File: libraries/AP_ADSB/AP_ADSB.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/AP_ADSB/AP_ADSB.cpp:154]: oOoOOOOo Thank you!

File: ArduPlane/ArduPlane.cpp (hunk 1 above)
Comment (magicrub) [reply ArduPlane/ArduPlane.cpp:269]: the aircraft is still expected to fly as expected at that min airspeed, it's wen you get below it that it does drastic things while it still has flight-surface control.

File: libraries/AP_ADSB/AP_ADSB.cpp (hunk 2 above)
Comment (magicrub) [reply libraries/AP_ADSB/AP_ADSB.cpp]: fixed

File: ArduPlane/ArduPlane.cpp (hunk 1 above)
Comment (magicrub) [reply ArduPlane/ArduPlane.cpp]: I've removed this

**Comment by magicrub** [comment]
rebased

**Comment by magicrub** [comment]
I just realized that I can use the serial manager protocol as the protocol option and remove this new param from adsb.

**Comment by magicrub** [comment]
I removed the new param and replaced it with a simple ::detect() call that checks for the already-required serial protocol setting. This means only 1 thing is needed to be set instead of 2 to enable this sensor. This is also an easy way-forward for Sagetech MX and uAvionix's GDL90/UDP drivers which are in the pipeline

**Comment by tridge** [comment]
this should not have been merged without proper review and testing

**Comment by magicrub** [comment]
@tridge see PR https://github.com/ArduPilot/ardupilot/pull/15517 for revert.
